mgmt/imgmgr/src/imgmgr.c

```diff
@@ -743,6 +743,45 @@ imgr_upload_good_rsp(struct mgmt_cbuf *cb)
     return 0;
 }
 
+/**
+ * Logs an upload request if necessary.
+ *
+ * @param is_first              Whether the request includes the first chunk of
+ *                                  the image.
+ * @param is_last               Whether the request includes the last chunk of
+ *                                  the image.
+ * @param status                The result of processing the upload request
+ *                                  (MGMT_ERR code).
+ *
+ * @return                      0 on success; nonzero on failure.
+ */
+static int
+imgr_upload_log(bool is_first, bool is_last, int status)
+{
+    uint8_t hash[IMGMGR_HASH_LEN];
+    const uint8_t *hashp;
+    int rc;
+
+    if (is_first) {
+        return imgmgr_log_upload_start(status);
+    }
+
+    /* Log the image hash if we know it. */
+    rc = imgr_read_info(1, NULL, hash, NULL);
```

Comment:
I would recommend moving this code behind the if-statement where you decide that this is going to log something. imgr_read_info() does quite a bit of flash activity, so it is a bit heavy to run for every data chunk.

Reply:
Thanks @mkiiskila, that is a good point.  I have changed the function as follows:
```
static int
imgr_upload_log(bool is_first, bool is_last, int status)
{
    uint8_t hash[IMGMGR_HASH_LEN];
    const uint8_t *hashp;
    int rc;

    if (is_first) {
        return imgmgr_log_upload_start(status);
    }

    if (is_last || status != 0) {
        /* Log the image hash if we know it. */
        rc = imgr_read_info(1, NULL, hash, NULL);
        if (rc != 0) {
            hashp = NULL;
        } else {
            hashp = hash;
        }

        return imgmgr_log_upload_done(status, hashp);
    }

    /* Nothing to log. */
    return 0;
}
```